openedx_lti_tool_plugin/models.py

```diff
@@ -208,31 +208,49 @@ def __str__(self) -> str:
         return f'<LtiProfile, ID: {self.id}>'
 
 
-class CourseAccessConfiguration(models.Model):
-    """Course access configuration.
+class LtiToolConfiguration(models.Model):
+    """LTI Tool Configuration.
 
-    A model to store LTI tool course access configuration.
+    A model to store LTI tool configuration.
     """
 
     EXAMPLE_ID_LIST = 'Example: ["id-1", "id-2", ...]'
 
+    NEW_ACCOUNTS_ONLY = 'new_only'
+    EXISTING_AND_NEW = 'existing_and_new'
+    EXISTING_ONLY = 'existing_only'
+
+    USER_PROVISIONING_MODE_CHOICES = [
+        (NEW_ACCOUNTS_ONLY, _('New accounts only.')),
+        (EXISTING_AND_NEW, _('Existing and new accounts.')),
+        (EXISTING_ONLY, _('Existing accounts only.')),
+    ]
+
     lti_tool = models.OneToOneField(
         LtiTool,
         on_delete=models.CASCADE,
-        related_name='openedx_lti_tool_plugin_course_permission',
+        related_name='openedx_lti_tool_plugin_configuration',
         verbose_name=_('LTI Tool'),
     )
     allowed_course_ids = models.TextField(
         verbose_name=_('Allowed Course IDs'),
         help_text=_(f'List of allowed courses IDs. {EXAMPLE_ID_LIST}'),
         default=[],
     )
+    user_provisioning_mode = models.CharField(
+        max_length=50,
+        choices=USER_PROVISIONING_MODE_CHOICES,
```

Comment:
@Serafin-dev What do you think about using enumeration type for the field choices?: https://docs.djangoproject.com/en/3.2/ref/models/fields/#enumeration-types